pex/cli/commands/lock.py

```diff
@@ -491,22 +486,37 @@ def _update(self):
                         file=output,
                     )
         if performed_update:
+            original_locked_project_names = {
+                locked_requirement.pin.project_name
+                for locked_resolve in lock_file.locked_resolves
+                for locked_requirement in locked_resolve.locked_requirements
+            }
+            new_requirements = OrderedSet(
```

Comment:
Erm, that's a bug when, for example, there was `-p foo` that was new, but it also introduces 10 other new transitive dependencies. Only the `foo` requirement should be added to the existing top-level requirements list.

Reply:
Actually, this is right. I'll push up a test that confirms.